Source/Core/VideoCommon/PixelEngine.cpp

```diff
@@ -273,26 +251,22 @@ static void SetFinish_OnMainThread(u64 userdata, s64 cyclesLate)
 // THIS IS EXECUTED FROM VIDEO THREAD
 void SetToken(const u16 _token, const int _bSetTokenAcknowledge)
 {
+  INFO_LOG(PIXELENGINE, "VIDEO Backend raises INT_CAUSE_PE_TOKEN (btw, token: %04x)", _token);
+  s_Token.store(_token);
```

Comment:
You could make a guarantee that the tokens sent with an interrupt will actually show up in the token register for X cycles after the interrupt.